azurerm/resource_arm_sql_database.go

```diff
@@ -118,128 +119,157 @@ func resourceArmSqlDatabase() *schema.Resource {
-	if !createResponse.IsSuccessful() {
-		return fmt.Errorf("Error creating SQL Database: %s", createResponse.Error)
+
+	parameters := sql.Database{
+		Name:               &databaseName,
+		DatabaseProperties: &props,
+		Tags:               metadata,
+		Location:           &location,
 	}
 
-	readRequest := rivieraClient.NewRequest()
-	readRequest.Command = &sql.GetDatabase{
-		Name:              d.Get("name").(string),
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ServerName:        d.Get("server_name").(string),
+	resultChan, errChan := sqlDatabasesClient.CreateOrUpdate(resGroup, serverName, databaseName, parameters, make(chan struct{}))
+	resultServer := <-resultChan
+	errServer := <-errChan
+	if errServer != nil {
+		return errServer
 	}
 
-	readResponse, err := readRequest.Execute()
-	if err != nil {
-		return fmt.Errorf("Error reading SQL Database: %s", err)
+	if resultServer.StatusCode != http.StatusOK {
+		return fmt.Errorf("Cannot create Sql database %s (resource group %s) ID", databaseName, resGroup)
 	}
-	if !readResponse.IsSuccessful() {
-		return fmt.Errorf("Error reading SQL Database: %s", readResponse.Error)
+
+	resultDb, errDb := sqlDatabasesClient.Get(resGroup, serverName, databaseName, "")
+	if errDb != nil {
+		d.SetId("")
+		return fmt.Errorf("Error reading SQL database %s: %v", databaseName, errDb)
 	}
 
-	resp := readResponse.Parsed.(*sql.GetDatabaseResponse)
-	d.SetId(*resp.ID)
+	d.SetId(*resultDb.ID)
 
 	return resourceArmSqlDatabaseRead(d, meta)
 }
 
 func resourceArmSqlDatabaseRead(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
+	sqlDatabasesClient := meta.(*ArmClient).sqlDatabasesClient
 
-	readRequest := rivieraClient.NewRequestForURI(d.Id())
-	readRequest.Command = &sql.GetDatabase{}
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
 
-	readResponse, err := readRequest.Execute()
+	resGroup := id.ResourceGroup
+	serverName := id.Path["servers"]
+	databaseName := id.Path["databases"]
+
+	result, err := sqlDatabasesClient.Get(resGroup, serverName, databaseName, "")
 	if err != nil {
-		return fmt.Errorf("Error reading SQL Database: %s", err)
+		return fmt.Errorf("Error reading SQL database %s: %v", databaseName, err)
 	}
-	if !readResponse.IsSuccessful() {
-		log.Printf("[INFO] Error reading SQL Database %q - removing from state", d.Id())
+	if result.Response.StatusCode == http.StatusNotFound {
 		d.SetId("")
-		return fmt.Errorf("Error reading SQL Database: %s", readResponse.Error)
+		return nil
 	}
 
-	resp := readResponse.Parsed.(*sql.GetDatabaseResponse)
+	databaseProperties := *result.DatabaseProperties
 
-	d.Set("name", resp.Name)
-	d.Set("creation_date", resp.CreationDate)
-	d.Set("default_secondary_location", resp.DefaultSecondaryLocation)
-	d.Set("elastic_pool_name", resp.ElasticPoolName)
+	d.Set("name", databaseName)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", *result.Location)
```

Comment:
can we change this to be `azureRMNormalizeLocation(*resp.Location)` to store it in the state correctly?